app/goal_routes.py

```diff
@@ -0,0 +1,97 @@
+
+from app import db
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response,request, abort
+from dotenv import load_dotenv
+import os
+from app.models.task import Task
+
+load_dotenv()
+goal_bp = Blueprint("goal", __name__,url_prefix="/goals")
+
+#Helper function
+def valid_int(number):
+    try:
+        return int(number)     
+    except:
+        abort(make_response({"error": f"{number} must be an int"}, 400))
+#Helper function
+def get_goal_from_id(goal_id):
+    goal_id = valid_int(goal_id)
+    return Goal.query.get_or_404(goal_id, description="{Goal not found}")
```

Comment:
Since `get_or_404` returns HTML by default you might want to do something like this instead:

```suggestion
    goal = Goal.query.get(goal_id)
    if goal:
        return goal
    else:
        abort(make_response({"description": "Goal not found"}, 404))
```